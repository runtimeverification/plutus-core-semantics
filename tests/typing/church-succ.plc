(program 1.0.0
  (lam
    n
    (all r (type) (fun r (fun (fun r r) r)))
    (abs r (type) (lam z r (lam f (fun r r) [ f [ [ { n r } z ] f ] ])))
  )
)