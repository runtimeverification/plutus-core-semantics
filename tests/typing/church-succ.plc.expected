<generatedTop>
  <k>
    ( fun ( all r ( type ) ( fun r ( fun ( fun r r ) r ) ) ) ( all r ( type ) ( fun r ( fun ( fun r r ) r ) ) ) ) :: ( type )
  </k>
  <env>
    .
  </env>
</generatedTop>
